modules/rostests/apitests/shell32/ShellExecCmdLine.cpp

```diff
@@ -626,8 +587,32 @@ static void DoEntry(const TEST_ENTRY *pEntry)
 
     ok(result == pEntry->result, "Line %d: result expected %d, was %d\n",
        pEntry->lineno, pEntry->result, result);
+}
+
+static WINDOW_LIST s_List1, s_List2;
+
+static VOID TEST_ShellExecCmdLine(VOID)
+{
+    GetWindowList(&s_List1);
 
-    CleanupNewlyCreatedWindows();
+    // do tests
+    for (size_t i = 0; i < _countof(s_entries_1); ++i)
+    {
+        DoEntry(&s_entries_1[i]);
+    }
+    SetEnvironmentVariableW(L"PATH", s_cur_dir);
+    for (size_t i = 0; i < _countof(s_entries_2); ++i)
+    {
+        DoEntry(&s_entries_2[i]);
+    }
+
+    Sleep(2000);
+
+    // Clean up
```

Comment:
Would it be possible to clean up directly after each affected entry? Or do we have to rely on such global handling?

Reply:
Impossible for `ShellExecCmdLine`.